prettify-ts-plugin/src/pluginCreator.ts

```diff
@@ -0,0 +1,121 @@
+import type * as ts from 'typescript'
+import type * as tsserver from 'typescript/lib/tsserverlibrary'
+
+import { Plugin } from './plugin'
+
+export interface PluginConfig {
+  /**
+     * Enable or disable the Prettify TS hover tooltip.
+     * @defaultValue `true`
+     */
+  enableHover: boolean
+
+  /**
+     * Show or hide nested types.
+     * @defaultValue `false`
+     */
+  viewNestedTypes: boolean
+
+  /**
+     * Indentations for type previews.
+     * @defaultValue `4`
+     */
+  typeIndentation: number
+
+  /**
+     * Types that won't be expanded when viewing deeply nested types. These types need to be in the global or local scope, or else the plugin will fail.
+     * @defaultValue `["String", "Number", "Boolean", "Date", "RegExp", "Function", "Symbol"]`
+     */
+  ignoredNestedTypes: string[]
```

Comment:
Two things here. First, the JSDoc comments look weird with Standard's formatting, at least to me.

Secondly and more complexly, I think that an interesting alternative or complementary setting might be `expandLibTypes`. I can't think of a perfect name for the setting but TypeScript ships with a library of type definitions for built in types like `String`, `Number`, etc. and that seems to be a part of goal here. See the [tsconfig setting](https://www.typescriptlang.org/tsconfig#lib).

I think that this achieves the goal much more thoroughly. Then next it might be interesting to have a setting to have `expandDependencyTypes` as well. I assume the default here would be `true` but it isn't necessarily as helpful to delve all the way into a dependency's type.

Honestly I began to realize some of the things I was looking to implement already exist though at: https://github.com/mxsdev/ts-type-explorer